File: frontend/src/types/contentSeries.ts
export type SeriesStatus = 'ACTIVE' | 'PAUSED' | 'COMPLETED' | 'ARCHIVED'

export interface SeriesEpisode {
  id: number
  seriesId: number
  episodeNumber: number
  title: string
  videoId?: string
  platform: string
  status: 'PLANNED' | 'DRAFTED' | 'PUBLISHED' | 'SKIPPED'
  scheduledDate?: string
  publishedDate?: string
  views?: number
  likes?: number
  comments?: number
}

export interface ContentSeries {
  id: number
  title: string
  description: string
  coverImageUrl?: string
  status: SeriesStatus
  platform: string
  frequency: 'DAILY' | 'WEEKLY' | 'BIWEEKLY' | 'MONTHLY' | 'CUSTOM'
  customFrequencyDays?: number
  totalEpisodes: number
  publishedEpisodes: number
  avgViews: number
  avgRetention: number
  totalViews: number
  episodes: SeriesEpisode[]
  tags: string[]
  createdAt: string
  updatedAt: string
}

export interface SeriesAnalytics {
  seriesId: number
  viewsTrend: { episode: number; views: number; retention: number }[]
  subscriberGrowth: number
  avgEngagement: number
  bestPerformingEpisode: SeriesEpisode | null
  dropOffRate: number
  audienceReturnRate: number
}

export interface CreateSeriesRequest {
  title: string
  description: string
  platform: string
  frequency: ContentSeries['frequency']
  plannedEpisodes?: number
  tags?: string[]
}

File: backend/onGo-domain/src/main/kotlin/com/ongo/domain/contentseries/ContentSeries.kt
package com.ongo.domain.contentseries

import java.time.LocalDateTime

data class ContentSeries(
    val id: Long? = null,
    val userId: Long,
    val title: String,
    val description: String? = null,
    val coverImageUrl: String? = null,
    val status: String = "ACTIVE",
    val platform: String = "YOUTUBE",
    val frequency: String = "WEEKLY",
    val customFrequencyDays: Int? = null,
    val tags: String = "[]",
    val createdAt: LocalDateTime? = null,
    val updatedAt: LocalDateTime? = null,
)
